stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/common/DateFormat.kt

```diff
@@ -0,0 +1,42 @@
+package io.getstream.chat.android.compose.ui.common
+
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import com.getstream.sdk.chat.utils.DateFormatter
+import com.getstream.sdk.chat.utils.formatDate
+import com.getstream.sdk.chat.utils.formatTime
+import io.getstream.chat.android.compose.state.DateFormatType
+import io.getstream.chat.android.compose.state.DateFormatType.DATE
+import io.getstream.chat.android.compose.state.DateFormatType.TIME
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import java.util.Date
+
+/**
+ * Represents a timestamp in the app, that's used primarily for channels and messages.
+ *
+ * @param date - The date to show in the timestamp.
+ * @param modifier - Modifier for styling.
+ * @param formatter - The formatter that's used to format the date to a String.
+ * @param formatType - The type of formatting to provide - either a timestamp or a full date. We format the information
+ * using [DATE] by default, as it's the most common behavior.
+ * */
+@Composable
+public fun DateFormat(
```

Comment:
The name we landed on for this component is weird for me. I keep thinking of [`SimpleDateFormat`](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html), I think this is something that contains info on how to format a date, and not something that displays it on the UI.

I think the original `Timestamp` (?) was probably better, but let's see what other ideas we have for this.